docs/howto/data_management_framework/resources.rst

```diff
@@ -0,0 +1,136 @@
+    rsrc.set_field("desc", "Example table")
+
+    t = Table()
+    # Assume existing comma-separated values file "data.csv", with a header row, e.g.
+    #     Index,Pressure,Temperature
+    #     1,256.12,38.7
+    #     2,125.61,73.8
+    t.read_csv("data.csv")
+    t.add_to_resource(rsrc)
+
+Create a resource representing a publication and multiple CSV tables
+---------------------------------------------------------------------
+In this case, the publication is the source for 10 related CSV tables.
+Each table has a header, with optional units in square brackets, and then value rows:
+
+=====  ======= =============  ==============  ===================  ===========  ===========  ===================
+T [C]  P [bar] G0/RT H2O [-]  G0/RT NaCl [-]  A phi [(kg/mol^0.5]  B0 [kg/mol]  B1 [kg/mol]  10^3xC [(kg/mol)^2]
+=====  ======= =============  ==============  ===================  ===========  ===========  ===================
+0      1       -23.4638       -13.836         0.3767               0.0493       0.2462       2.58
+10     1       -22.91         -13.871         0.3821               0.0619       0.2612       1.68
+...    ...     ...            ...             ...                  ...          ...          ...
+=====  ======= =============  ==============  ===================  ===========  ===========  ===================
+
+The source publication and the 10 datasets (CSV files) are each separate `resources` in the DMF, with a
+relationship of "derived [from]" linking them. This can be pictured like this::
+
+                  ┌───────────────┐
+                  │ pitzer-source │
+                  └──────▲────────┘
+                         │    derived from
+               ┌─────────┴────────┬─────────────────────┐
+               │                  │                     │
+        ┌──────┴───────┐  ┌───────┴─────┐      ┌────────┴─────┐
+        │ dataset1     │  │ dataset2    │      │ dataset10    │
+        │ pitzer_1.csv │  │ pitzer_2.csv│ ...  │ pitzer_10.csv│
+        └──────────────┘  └─────────────┘      └──────────────┘
+
+And finally, the code:
+
+.. code-block:: python
+
+    # imports
+    from idaes.dmf import DMF, resource, tables
+    from pathlib import Path
+
+    workspace_name = "pitzer_data"
+
+    # Create DMF workspace to store data
+    dmf = DMF(workspace_name, create=True)
+
+    # Create/add resource representing source publication
+    src = resource.Resource(name="pitzer-source", type_=resource.ResourceTypes.publication)
+    src.v["sources"].append({"date": "1984",
+                       "doi": "10.1063/1.555709",
+                       "source": 'Pitzer, Kenneth S., J. Christopher Peiper, and R. H. Busey. '
+                       '"Thermodynamic properties of aqueous sodium chloride solutions." '
+                       'Journal of Physical and Chemical Reference Data 13.1 (1984): 1-102.'})
+    dmf.add(src)
+
+
+    # Find tables to read, i.e. look for CSV files
+    data_files = Path(".").glob("*.csv")
```

Comment:
To clarify, from reading this it would imply that I need to add the .csv files to the repo as the original data source, then load them into the DMF when I want to use them (i.e. in the test files somewhere).

Reply:
If you don't save the DMF after you load them, yes. If you save the DMF contents after they are loaded, you don't (in fact, shouldn't) do it again

Reply:
fwiw this is basically what you could say about loading data into any database

Reply:
@andrewlee94 coming back to this, I think *yes* we should have the data in the repo. I would argue there should be a top-level "data" directory that is also a DMF _workspace_, where data files are loaded. This will make the loading procedure unnecessary for the user, they will only need to fetch the data.